src/xSTIR/cSTIR/stir_data_containers.cpp

```diff
@@ -346,6 +346,45 @@ const DataContainer& a_y
 	}
 }
 
+STIRImageData::STIRImageData(const ImageData& id)
+{
+    // Copy the geometrical info
+    _geom_info_sptr = std::make_shared<VoxelisedGeometricalInfo3D>(*id.get_geom_info());
+
+    // Size
+    // TODO should this use geom info?
+    Dimensions dim = id.dimensions();
+    int nx = dim["x"];
+    int ny = dim["y"];
+    int nz = 1;
+    Dimensions::iterator it = dim.begin();
+    while (it != dim.end()) {
+        if (it->first != "x" && it->first != "y")
+            nz *= it->second;
+        ++it;
+    }
+    stir::IndexRange3D index_range(     0,          nz - 1,
+                                   -(ny / 2), -(ny / 2) + ny - 1,
+                                   -(nx / 2), -(nx / 2) + nx - 1);
+
+    // Spacing
+    VoxelisedGeometricalInfo3D::Spacing id_spacing = _geom_info_sptr->get_spacing();
+    Coord3DF spacing(id_spacing[2],id_spacing[1],id_spacing[2]);
+
+    // Offset
+    VoxelisedGeometricalInfo3D::Offset id_offset = _geom_info_sptr->get_offset();
+    Coord3DF offset(id_offset[2],id_offset[1],id_offset[2]);
```

Comment:
there is no reason to believe that the STIR offset and SIRF offset are the same. (they probably aren't)